src/transports/http.rs

```diff
@@ -128,14 +128,20 @@ impl Http {
             let url = Url::parse(url)?;
             let user = url.username();
 
-            if !user.is_empty() {
-                let auth = match url.password() {
-                    Some(pass) => format!("{}:{}", user, pass),
-                    None => format!("{}:", user),
-                };
-                Some(HeaderValue::from_str(&format!("Basic {}", base64::encode(&auth)))?)
-            } else {
+            let auth = match url.password() {
```

Comment:
```rust
let auth = format("{}:{}", user, url.password().unwrap_or_default());
if auth == ":" {
  None 
} else {
  Some(...)
}
```

Should work the same and feels less complex to me.

Reply:
reduced complexity as you suggested. Thank you for the feedback and approval. This will make Infura project secret support much easier in our application.